markdown.go

```diff
@@ -213,14 +213,16 @@ func (p *parser) finalize(block *Node) {
 }
 
 func (p *parser) addChild(node NodeType, offset uint32) *Node {
-	for !p.tip.canContain(node) {
+	return p.addExistingChild(NewNode(node), offset)
```

Comment:
There are only 3 callers of `addChild` now, and it differs from `addExistingChild` only by `NewNode`.

See if it's better to keep it or remove it. Up to you, either is fine. Just wanted to raise your attention here.